app/models/course/material/folder.rb

```diff
@@ -35,4 +55,14 @@ def validate_name_is_unique_among_materials
     conflicts = parent.materials.where('lower(name) = ?', name.downcase)
     errors.add(:name, :taken) if conflicts.any?
   end
+
+  def existing_names
+    parent.children.where.not(id: self).pluck(:name) + parent.materials.pluck(:name)
```

Comment:
this is OK, but if we have a similar query again in future we should have a method called siblings.